Sources/Danger/Plugins/SwiftLint/SwiftLint.swift

```diff
@@ -4,42 +4,81 @@ import Foundation
 /// it usable out of the box.
 public enum SwiftLint {
+    public enum LintStyle {
+        /// Lints all the files instead of only the modified and created files.
+        /// - Parameters:
+        ///   - directory: Optional property to set the --path to execute at.
+        case all(directory: String?)
+
+        /// Only lints the modified and created files with `.swift` extension.
+        /// - Parameters:
+        ///   - directory: Optional property to set the --path to execute at.
+        case modifiedAndCreatedFiles(directory: String?)
+
+        /// Lints only the given files. This can be useful to do some manual filtering.
+        /// The files will be filtered on `.swift` only.
+        case files([File])
+    }
+
     static let danger = Danger()
     static let shellExecutor = ShellExecutor()
 
+    /// This method is deprecated in favor of
+    @available(*, deprecated, message: "Use the lint(_ lintStyle ..) method instead.")
+    @discardableResult
+    public static func lint(inline: Bool = false,
+                            directory: String? = nil,
+                            configFile: String? = nil,
+                            strict: Bool = false,
+                            quiet: Bool = true,
+                            lintAllFiles: Bool = false,
+                            swiftlintPath: String? = nil) -> [SwiftLintViolation] {
+        let lintStyle: LintStyle = lintAllFiles ? .all(directory: directory) : .modifiedAndCreatedFiles(directory: directory)
+        return lint(lintStyle,
+                    inline: inline,
+                    configFile: configFile,
+                    strict: strict,
+                    quiet: quiet,
+                    lintAllFiles: lintAllFiles,
+                    swiftlintPath: swiftlintPath)
+    }
+
     /// This is the main entry point for linting Swift in PRs.
     ///
     /// When the swiftlintPath is not specified,
     /// it uses by default swift run swiftlint if the Package.swift contains swiftlint as dependency,
     /// otherwise calls directly the swiftlint command
-
     @discardableResult
-    public static func lint(inline: Bool = false, directory: String? = nil,
-                            configFile: String? = nil, strict: Bool = false, lintAllFiles: Bool = false,
+    public static func lint(_ lintStyle: LintStyle = .modifiedAndCreatedFiles(directory: nil),
+                            inline: Bool = false,
+                            configFile: String? = nil,
+                            strict: Bool = false,
+                            quiet: Bool = true,
+                            lintAllFiles: Bool = false,
                             swiftlintPath: String? = nil) -> [SwiftLintViolation] {
-        lint(danger: danger,
+        lint(lintStyle: lintStyle,
+             danger: danger,
              shellExecutor: shellExecutor,
              swiftlintPath: swiftlintPath ?? SwiftLint.swiftlintDefaultPath(),
              inline: inline,
```

Comment:
<!--
  0 failure: 
  1 warning:  Function should h...
  
  
  DangerID: danger-id-Danger;
  File: Sources/Danger/Plugins/SwiftLint/SwiftLint.swift;
  Line: 63;
-->

- :warning: Function should have complexity 10 or less: currently complexity equals 11 (`cyclomatic_complexity`)